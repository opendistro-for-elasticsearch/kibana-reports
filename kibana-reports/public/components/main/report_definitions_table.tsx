/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React, { useState } from 'react';
import {
  EuiLink,
  EuiInMemoryTable,
  EuiButton,
  EuiEmptyPrompt,
  EuiText,
  EuiIcon,
} from '@elastic/eui';
import { humanReadableDate } from './main_utils';

const emptyMessageReportDefinitions = (
  <EuiEmptyPrompt
    title={<h3>No report definitions to display</h3>}
    titleSize="xs"
    body={
      <div>
        <EuiText>Create a new report definition to get started</EuiText>
        <EuiText>
          To learn more, see{' '}
          <EuiLink>
            Get started with Kibana reporting <EuiIcon type="popout" />
          </EuiLink>
        </EuiText>
      </div>
    }
    actions={
      <div>
        <EuiButton
          onClick={() => {
            window.location.assign('opendistro-kibana-reports#/create');
          }}
        >
          Create report definition
        </EuiButton>
      </div>
    }
  />
);

const reportDefinitionsSearch = {
  box: {
    incremental: true,
  },
  filters: [],
};

export function ReportDefinitions(props) {
  const { pagination, getRowProps, reportDefinitionsTableContent } = props;

  const [sortField, setSortField] = useState('lastUpdated');
  const [sortDirection, setSortDirection] = useState('des');

  const sorting = {
    sort: {
      field: sortField,
      direction: sortDirection,
    },
  };

  const getDefinitionTableItemId = (name) => {
    let index;
    for (
      index = 0;
      index < props.reportDefinitionsTableContent.length;
      ++index
    ) {
      if (name === reportDefinitionsTableContent[index].reportName) {
        return reportDefinitionsTableContent[index].id;
      }
    }
  };

  const navigateToDefinitionDetails = (name: any) => {
    let id = getDefinitionTableItemId(name);
    window.location.assign(
      `opendistro_kibana_reports#/report_definition_details/${id}`
    );
  };

  const reportDefinitionsColumns = [
    {
      field: 'reportName',
      name: 'Name',
      render: (name) => (
        <EuiLink onClick={() => navigateToDefinitionDetails(name)}>
          {name}
        </EuiLink>
      ),
    },
    {
      field: 'type',
      name: 'Type',
      sortable: true,
      truncateText: false,
    },
    {
      field: 'owner',
      name: 'Owner',
      sortable: true,
      truncateText: false,
    },
    {
      field: 'source',
      name: 'Source',
      render: (username: string) => (
        <EuiLink href={'#'} target="_blank">
          {username}
        </EuiLink>
      ),
    },
    {
      field: 'lastUpdated',
      name: 'Last Updated',
      render: (date) => {
        let readable = humanReadableDate(date);
        return <EuiText size="s">{readable}</EuiText>;
      },
    },
    {
      field: 'details',
      name: 'Details',
      sortable: false,
      truncateText: true,
    },
    {
      field: 'status',
      name: 'Status',
      sortable: true,
      truncateText: false,
    },
  ];

  return (
    <div>
      <EuiInMemoryTable
        items={reportDefinitionsTableContent}
        itemId="id"
        loading={false}
        message={emptyMessageReportDefinitions}
        columns={reportDefinitionsColumns}
        search={reportDefinitionsSearch}
        pagination={pagination}
        sorting={sorting}
        isSelectable={true}
        tableLayout={'auto'}
      />
    </div>
  );
}
