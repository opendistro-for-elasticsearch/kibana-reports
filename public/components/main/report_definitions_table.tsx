/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import {
  EuiLink,
  EuiInMemoryTable,
  EuiButton,
  EuiEmptyPrompt,
  EuiText,
} from '@elastic/eui';
import { report_definitions } from './test_data';

export const scheduled_report_columns = [
  {
    field: 'reportName',
    name: 'Name',
    sortable: true,
    truncateText: true,
  },
  {
    field: 'type',
    name: 'Type',
    sortable: true,
    truncateText: false,
  },
  {
    field: 'owner',
    name: 'Owner',
    sortable: true,
    truncateText: false,
  },
  {
    field: 'source',
    name: 'Source',
    render: (username: string) => (
      <EuiLink href={'#'} target="_blank">
        {username}
      </EuiLink>
    ),
  },
  {
    field: 'lastUpdated',
    name: 'Last Updated',
    truncateText: true,
  },
  {
    field: 'details',
    name: 'Details',
    sortable: false,
    truncateText: true,
  },
  {
    field: 'status',
    name: 'Status',
    sortable: true,
    truncateText: false,
  },
];

const empty_message_report_definitions = (
  <EuiEmptyPrompt
    title={<h3>No report definitions to display</h3>}
    titleSize="xs"
    body={
      <div>
        <EuiText>Create a new report definition to get started</EuiText>
        <EuiText>
          To learn more, see{' '}
          <EuiLink>Get started with Kibana reporting</EuiLink>
        </EuiText>
      </div>
    }
    actions={
      <div>
        <EuiButton
          onClick={() => {
            window.location.assign('opendistro-kibana-reports#/create');
          }}
        >
          Create
        </EuiButton>
      </div>
    }
  />
);

const report_definitions_search = {
  box: {
    incremental: true,
  },
  filters: [],
};

export function ReportDefinitions(props) {
  const { pagination, getRowProps } = props;
  return (
    <div>
      <EuiInMemoryTable
        items={report_definitions}
        itemId="id"
        loading={false}
        message={empty_message_report_definitions}
        columns={scheduled_report_columns}
        search={report_definitions_search}
        pagination={pagination}
        sorting={true}
        isSelectable={true}
        rowProps={getRowProps}
      />
    </div>
  );
}
