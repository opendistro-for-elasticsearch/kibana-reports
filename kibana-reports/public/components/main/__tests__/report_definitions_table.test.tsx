/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { ReportDefinitions } from '../report_definitions_table';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

const pagination = {
  initialPageSize: 10,
  pageSizeOptions: [8, 10, 13],
};

describe('<ReportDefinitions /> panel', () => {
  configure({ adapter: new Adapter() });
  test('render component', () => {
    let reportDefinitionsTableContent = [
      {
        reportName: 'test report name',
        type: 'Download',
        owner: 'davidcui',
        source: 'Dashboard',
        lastUpdated: 'test updated time',
        details: '',
        status: 'Created',
      },
      {
        reportName: 'test report name 2',
        type: 'Download',
        owner: 'davidcui',
        source: 'Dashboard',
        lastUpdated: 'test updated time',
        details: '',
        status: 'Created',
      },
    ];
    const { container } = render(
      <ReportDefinitions
        pagination={pagination}
        reportDefinitionsTableContent={reportDefinitionsTableContent}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('render empty table', () => {
    const { container } = render(
      <ReportDefinitions
        pagination={pagination}
        reportDefinitionsTableContent={[]}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('test click on report definition row', async () => {
    window = Object.create(window);
    Object.defineProperty(window, 'location', {
      configurable: true,
      value: {
        assign: jest.fn(),
      },
    });
    let promise = Promise.resolve();
    let reportDefinitionsTableContent = [
      {
        reportName: 'test report name',
        type: 'Download',
        owner: 'davidcui',
        source: 'Dashboard',
        lastUpdated: 'test updated time',
        details: '',
        status: 'Created',
      },
      {
        reportName: 'test report name 2',
        type: 'Download',
        owner: 'davidcui',
        source: 'Dashboard',
        lastUpdated: 'test updated time',
        details: '',
        status: 'Created',
      },
    ];

    const component = mount(
      <ReportDefinitions
        pagination={pagination}
        reportDefinitionsTableContent={reportDefinitionsTableContent}
      />
    );

    const nameLink = component.find('button').at(3);
    nameLink.simulate('click');
  });
});
