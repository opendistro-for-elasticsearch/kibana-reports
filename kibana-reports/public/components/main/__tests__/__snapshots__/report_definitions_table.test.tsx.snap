// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportDefinitions /> panel render component 1`] = `
<div>
  <div>
    <div
      class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap"
    >
      <div
        class="euiFlexItem euiSearchBar__searchHolder"
      >
        <div
          class="euiFormControlLayout euiFormControlLayout--fullWidth"
        >
          <div
            class="euiFormControlLayout__childrenWrapper"
          >
            <input
              aria-label="This is a search bar. As you type, the results lower in the page will automatically filter."
              class="euiFieldSearch euiFieldSearch--fullWidth euiFieldSearch-isClearable"
              placeholder="Search..."
              type="search"
              value=""
            />
            <div
              class="euiFormControlLayoutIcons"
            >
              <span
                class="euiFormControlLayoutCustomIcon"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero euiSearchBar__filtersHolder"
      >
        <div
          class="euiFilterGroup"
        />
      </div>
    </div>
    <div
      class="euiSpacer euiSpacer--l"
    />
    <div
      class="euiBasicTable"
    >
      <div>
        <div
          class="euiTableHeaderMobile"
        >
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            />
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <div
                class="euiTableSortMobile"
              >
                <div
                  class="euiPopover euiPopover--anchorDownRight"
                >
                  <div
                    class="euiPopover__anchor"
                  >
                    <button
                      class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiButtonEmpty--flushRight"
                      type="button"
                    >
                      <span
                        class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                      >
                        <svg
                          aria-hidden="true"
                          class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonContent__icon"
                          focusable="false"
                          height="16"
                          role="img"
                          viewBox="0 0 16 16"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        />
                        <span
                          class="euiButtonEmpty__text"
                        >
                          Sorting
                        </span>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <table
          class="euiTable euiTable--responsive euiTable--auto"
          id="random_html_id"
          tabindex="-1"
        >
          <caption
            class="euiScreenReaderOnly euiTableCaption"
          />
          <thead>
            <tr>
              <th
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_reportName_0"
                role="columnheader"
                scope="col"
              >
                <div
                  class="euiTableCellContent"
                >
                  <span
                    class="euiTableCellContent__text"
                    title="Name"
                  >
                    Name
                  </span>
                </div>
              </th>
              <th
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_source_1"
                role="columnheader"
                scope="col"
              >
                <div
                  class="euiTableCellContent"
                >
                  <span
                    class="euiTableCellContent__text"
                    title="Source"
                  >
                    Source
                  </span>
                </div>
              </th>
              <th
                aria-live="polite"
                aria-sort="none"
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_type_2"
                role="columnheader"
                scope="col"
              >
                <button
                  class="euiTableHeaderButton"
                  data-test-subj="tableHeaderSortButton"
                  type="button"
                >
                  <span
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Type"
                    >
                      Type
                    </span>
                    <span
                      class="euiScreenReaderOnly"
                    >
                      Click to sort in ascending order
                    </span>
                  </span>
                </button>
              </th>
              <th
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_details_3"
                role="columnheader"
                scope="col"
              >
                <div
                  class="euiTableCellContent"
                >
                  <span
                    class="euiTableCellContent__text"
                    title="Schedule details"
                  >
                    Schedule details
                  </span>
                </div>
              </th>
              <th
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_lastUpdated_4"
                role="columnheader"
                scope="col"
              >
                <div
                  class="euiTableCellContent"
                >
                  <span
                    class="euiTableCellContent__text"
                    title="Last Updated"
                  >
                    Last Updated
                  </span>
                </div>
              </th>
              <th
                aria-live="polite"
                aria-sort="none"
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_status_5"
                role="columnheader"
                scope="col"
              >
                <button
                  class="euiTableHeaderButton"
                  data-test-subj="tableHeaderSortButton"
                  type="button"
                >
                  <span
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Status"
                    >
                      Status
                    </span>
                    <span
                      class="euiScreenReaderOnly"
                    >
                      Click to sort in ascending order
                    </span>
                  </span>
                </button>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              class="euiTableRow euiTableRow-isSelectable"
            >
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Name
                </div>
                <div
                  class="euiTableCellContent euiTableCellContent--overflowingContent"
                >
                  <button
                    class="euiLink euiLink--primary"
                    id="reportDefinitionDetailsLink"
                    type="button"
                  >
                    test report name
                  </button>
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Source
                </div>
                <div
                  class="euiTableCellContent euiTableCellContent--overflowingContent"
                >
                  <button
                    class="euiLink euiLink--primary"
                    type="button"
                  >
                    Dashboard
                  </button>
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Type
                </div>
                <div
                  class="euiTableCellContent"
                >
                  <span
                    class="euiTableCellContent__text"
                  >
                    Download
                  </span>
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Schedule details
                </div>
                <div
                  class="euiTableCellContent euiTableCellContent--truncateText"
                >
                  <span
                    class="euiTableCellContent__text"
                  />
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Last Updated
                </div>
                <div
                  class="euiTableCellContent euiTableCellContent--overflowingContent"
                >
                  <div
                    class="euiText euiText--small"
                  >
                    Invalid Date @ Invalid Date
                  </div>
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Status
                </div>
                <div
                  class="euiTableCellContent"
                >
                  <span
                    class="euiTableCellContent__text"
                  >
                    Created
                  </span>
                </div>
              </td>
            </tr>
            <tr
              class="euiTableRow euiTableRow-isSelectable"
            >
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Name
                </div>
                <div
                  class="euiTableCellContent euiTableCellContent--overflowingContent"
                >
                  <button
                    class="euiLink euiLink--primary"
                    id="reportDefinitionDetailsLink"
                    type="button"
                  >
                    test report name 2
                  </button>
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Source
                </div>
                <div
                  class="euiTableCellContent euiTableCellContent--overflowingContent"
                >
                  <button
                    class="euiLink euiLink--primary"
                    type="button"
                  >
                    Dashboard
                  </button>
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Type
                </div>
                <div
                  class="euiTableCellContent"
                >
                  <span
                    class="euiTableCellContent__text"
                  >
                    Download
                  </span>
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Schedule details
                </div>
                <div
                  class="euiTableCellContent euiTableCellContent--truncateText"
                >
                  <span
                    class="euiTableCellContent__text"
                  />
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Last Updated
                </div>
                <div
                  class="euiTableCellContent euiTableCellContent--overflowingContent"
                >
                  <div
                    class="euiText euiText--small"
                  >
                    Invalid Date @ Invalid Date
                  </div>
                </div>
              </td>
              <td
                class="euiTableRowCell"
              >
                <div
                  class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                >
                  Status
                </div>
                <div
                  class="euiTableCellContent"
                >
                  <span
                    class="euiTableCellContent__text"
                  >
                    Created
                  </span>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div>
        <div
          class="euiSpacer euiSpacer--m"
        />
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <div
              class="euiPopover euiPopover--anchorUpRight euiPopover--withTitle"
            >
              <div
                class="euiPopover__anchor"
              >
                <button
                  class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall"
                  data-test-subj="tablePaginationPopoverButton"
                  type="button"
                >
                  <span
                    class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                  >
                    <svg
                      aria-hidden="true"
                      class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonContent__icon"
                      focusable="false"
                      height="16"
                      role="img"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                    <span
                      class="euiButtonEmpty__text"
                    >
                      Rows per page
                      : 
                      10
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <nav
              class="euiPagination"
            >
              <button
                aria-label="Previous page"
                class="euiButtonIcon euiButtonIcon--text"
                data-test-subj="pagination-button-previous"
                disabled=""
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonIcon__icon"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </button>
              <ul
                class="euiPagination__list"
              >
                <li
                  class="euiPagination__item"
                >
                  <button
                    aria-controls="random_html_id"
                    aria-current="true"
                    aria-label="Page 1 of 1"
                    class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall euiButtonEmpty-isDisabled euiPaginationButton euiPaginationButton-isActive euiPaginationButton--hideOnMobile"
                    data-test-subj="pagination-button-0"
                    disabled=""
                    type="button"
                  >
                    <span
                      class="euiButtonContent euiButtonEmpty__content"
                    >
                      <span
                        class="euiButtonEmpty__text"
                      >
                        1
                      </span>
                    </span>
                  </button>
                </li>
              </ul>
              <button
                aria-label="Next page"
                class="euiButtonIcon euiButtonIcon--text"
                data-test-subj="pagination-button-next"
                disabled=""
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonIcon__icon"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </button>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ReportDefinitions /> panel render empty table 1`] = `
<div>
  <div>
    <div
      class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap"
    >
      <div
        class="euiFlexItem euiSearchBar__searchHolder"
      >
        <div
          class="euiFormControlLayout euiFormControlLayout--fullWidth"
        >
          <div
            class="euiFormControlLayout__childrenWrapper"
          >
            <input
              aria-label="This is a search bar. As you type, the results lower in the page will automatically filter."
              class="euiFieldSearch euiFieldSearch--fullWidth euiFieldSearch-isClearable"
              placeholder="Search..."
              type="search"
              value=""
            />
            <div
              class="euiFormControlLayoutIcons"
            >
              <span
                class="euiFormControlLayoutCustomIcon"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.271 11.978l3.872 3.873a.502.502 0 00.708 0 .502.502 0 000-.708l-3.565-3.564c2.38-2.747 2.267-6.923-.342-9.532-2.73-2.73-7.17-2.73-9.898 0-2.728 2.729-2.728 7.17 0 9.9a6.955 6.955 0 004.949 2.05.5.5 0 000-1 5.96 5.96 0 01-4.242-1.757 6.01 6.01 0 010-8.486c2.337-2.34 6.143-2.34 8.484 0a6.01 6.01 0 010 8.486.5.5 0 00.034.738z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero euiSearchBar__filtersHolder"
      >
        <div
          class="euiFilterGroup"
        />
      </div>
    </div>
    <div
      class="euiSpacer euiSpacer--l"
    />
    <div
      class="euiBasicTable"
    >
      <div>
        <div
          class="euiTableHeaderMobile"
        >
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            />
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <div
                class="euiTableSortMobile"
              >
                <div
                  class="euiPopover euiPopover--anchorDownRight"
                >
                  <div
                    class="euiPopover__anchor"
                  >
                    <button
                      class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiButtonEmpty--flushRight"
                      type="button"
                    >
                      <span
                        class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                      >
                        <svg
                          aria-hidden="true"
                          class="euiIcon euiIcon--medium euiButtonContent__icon"
                          focusable="false"
                          height="16"
                          role="img"
                          viewBox="0 0 16 16"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M13.069 5.157L8.384 9.768a.546.546 0 01-.768 0L2.93 5.158a.552.552 0 00-.771 0 .53.53 0 000 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 000-.76.552.552 0 00-.771 0z"
                            fill-rule="non-zero"
                          />
                        </svg>
                        <span
                          class="euiButtonEmpty__text"
                        >
                          Sorting
                        </span>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <table
          class="euiTable euiTable--responsive euiTable--auto"
          id="random_html_id"
          tabindex="-1"
        >
          <caption
            class="euiScreenReaderOnly euiTableCaption"
          />
          <thead>
            <tr>
              <th
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_reportName_0"
                role="columnheader"
                scope="col"
              >
                <div
                  class="euiTableCellContent"
                >
                  <span
                    class="euiTableCellContent__text"
                    title="Name"
                  >
                    Name
                  </span>
                </div>
              </th>
              <th
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_source_1"
                role="columnheader"
                scope="col"
              >
                <div
                  class="euiTableCellContent"
                >
                  <span
                    class="euiTableCellContent__text"
                    title="Source"
                  >
                    Source
                  </span>
                </div>
              </th>
              <th
                aria-live="polite"
                aria-sort="none"
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_type_2"
                role="columnheader"
                scope="col"
              >
                <button
                  class="euiTableHeaderButton"
                  data-test-subj="tableHeaderSortButton"
                  type="button"
                >
                  <span
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Type"
                    >
                      Type
                    </span>
                    <span
                      class="euiScreenReaderOnly"
                    >
                      Click to sort in ascending order
                    </span>
                  </span>
                </button>
              </th>
              <th
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_details_3"
                role="columnheader"
                scope="col"
              >
                <div
                  class="euiTableCellContent"
                >
                  <span
                    class="euiTableCellContent__text"
                    title="Schedule details"
                  >
                    Schedule details
                  </span>
                </div>
              </th>
              <th
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_lastUpdated_4"
                role="columnheader"
                scope="col"
              >
                <div
                  class="euiTableCellContent"
                >
                  <span
                    class="euiTableCellContent__text"
                    title="Last Updated"
                  >
                    Last Updated
                  </span>
                </div>
              </th>
              <th
                aria-live="polite"
                aria-sort="none"
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_status_5"
                role="columnheader"
                scope="col"
              >
                <button
                  class="euiTableHeaderButton"
                  data-test-subj="tableHeaderSortButton"
                  type="button"
                >
                  <span
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Status"
                    >
                      Status
                    </span>
                    <span
                      class="euiScreenReaderOnly"
                    >
                      Click to sort in ascending order
                    </span>
                  </span>
                </button>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              class="euiTableRow"
            >
              <td
                class="euiTableRowCell euiTableRowCell--isMobileFullWidth"
                colspan="6"
              >
                <div
                  class="euiTableCellContent euiTableCellContent--alignCenter"
                >
                  <span
                    class="euiTableCellContent__text"
                  >
                    <div
                      class="euiEmptyPrompt"
                    >
                      <span
                        class="euiTextColor euiTextColor--subdued"
                      >
                        <h3
                          class="euiTitle euiTitle--xsmall"
                        >
                          No report definitions to display
                        </h3>
                        <div
                          class="euiSpacer euiSpacer--m"
                        />
                        <div
                          class="euiText euiText--medium"
                        >
                          <div>
                            <div
                              class="euiText euiText--medium"
                            >
                              Create a new report definition to get started
                            </div>
                            <div
                              class="euiText euiText--medium"
                            >
                              To learn more, see
                               
                              <button
                                class="euiLink euiLink--primary"
                                type="button"
                              >
                                Get started with Kibana reporting 
                                <svg
                                  aria-hidden="true"
                                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                                  focusable="false"
                                  height="16"
                                  role="img"
                                  viewBox="0 0 16 16"
                                  width="16"
                                  xmlns="http://www.w3.org/2000/svg"
                                />
                              </button>
                            </div>
                          </div>
                        </div>
                      </span>
                      <div
                        class="euiSpacer euiSpacer--l"
                      />
                      <div
                        class="euiSpacer euiSpacer--s"
                      />
                      <div>
                        <button
                          class="euiButton euiButton--primary"
                          type="button"
                        >
                          <span
                            class="euiButtonContent euiButton__content"
                          >
                            <span
                              class="euiButton__text"
                            >
                              Create report definition
                            </span>
                          </span>
                        </button>
                      </div>
                    </div>
                  </span>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
